### PR TITLE
KTOR-6687 Empty CSRF onFailure option is misleading

### PR DESCRIPTION
**Pull request by bjhham**
**Subsystem**
Server, CSRF

**Motivation**
- KTOR-6687 onFailure without an explicit response would result in a 200 empty body

**Solution**
- Added the default error response when there is no response committed in the onFailure block

### CODE REVIEW

File: ktor-server/ktor-server-plugins/ktor-server-csrf/jvmAndNix/src/io/ktor/server/plugins/csrf/CSRF.kt [hunk 1]
```diff
@@ -38,6 +38,11 @@ public val CSRF: RouteScopedPlugin<CSRFConfig> = createRouteScopedPlugin("CSRF",
     val headerChecks = pluginConfig.headerChecks
     val onFailure = pluginConfig.handleFailure
 
+    if (!checkHost && allowedOrigins.isEmpty() && headerChecks.isEmpty()) {
```
Comment (Stexxe) [inline ktor-server/ktor-server-plugins/ktor-server-csrf/jvmAndNix/src/io/ktor/server/plugins/csrf/CSRF.kt:41]: Can you tell me how this code block is related to the onFailure problem?

Comment (bjhham) [reply ktor-server/ktor-server-plugins/ktor-server-csrf/jvmAndNix/src/io/ktor/server/plugins/csrf/CSRF.kt:41]: This is the second bullet in the description, to address that an empty configuration doesn't provide any functionality.  There's no real valid option for the default behaviour in this plugin so I'm proposing that we could log a warning when there's no validation defined.

Comment (Stexxe) [reply ktor-server/ktor-server-plugins/ktor-server-csrf/jvmAndNix/src/io/ktor/server/plugins/csrf/CSRF.kt:41]: Every commit should have a prefix with a YouTrack issue ID. I don't see the ID for the second commit.

Comment (Stexxe) [reply ktor-server/ktor-server-plugins/ktor-server-csrf/jvmAndNix/src/io/ktor/server/plugins/csrf/CSRF.kt:41]: Also, having a separate PR for an issue is a good practice.

Comment (bjhham) [reply ktor-server/ktor-server-plugins/ktor-server-csrf/jvmAndNix/src/io/ktor/server/plugins/csrf/CSRF.kt:41]: Sure, I'll move it to another 👍

**Review by e5l — APPROVED** [review]

**Pull request merged**